extensions/kubernetes/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java

```diff
@@ -92,20 +94,39 @@ public void build(ApplicationInfoBuildItem applicationInfo,
         Map<String, Object> configAsMap = StreamSupport.stream(config.getPropertyNames().spliterator(), false)
                 .filter(k -> ALLOWED_GENERATOR.equals(generatorName(k)))
                 .collect(Collectors.toMap(k -> PROPERTY_PREFIX + k, k -> config.getValue(k, String.class)));
+        // this is a hack to get kubernetes.registry working because currently it's not supported as is in Dekorate
+        Optional<String> kubernetesRegistry = config.getOptionalValue(ALLOWED_GENERATOR + ".registry", String.class);
+        if (kubernetesRegistry.isPresent()) {
+            System.setProperty(DOCKER_REGISTRY_PROPERTY, kubernetesRegistry.get());
+        }
+        // this is a hack to work around Dekorate using the default group for some of the properties
+        Optional<String> kubernetesGroup = config.getOptionalValue(ALLOWED_GENERATOR + ".group", String.class);
+        if (kubernetesGroup.isPresent()) {
+            System.setProperty(APP_GROUP_PROPERTY, kubernetesGroup.get());
+        }
 
         final SessionWriter sessionWriter = new SimpleFileWriter(root, false);
         Project project = createProject(applicationInfo, archiveRootBuildItem);
         sessionWriter.setProject(project);
         final Session session = Session.getSession();
         session.setWriter(sessionWriter);
-        session.feed(Maps.fromProperties(configAsMap));
 
+        session.feed(Maps.fromProperties(configAsMap));
         //apply build item configurations to the dekorate session.
         applyBuildItems(session, applicationInfo, kubernetesPortBuildItems, kubernetesHealthLivenessPathBuildItem,
                 kubernetesHealthReadinessPathBuildItem);
 
         // write the generated resources to the filesystem
         final Map<String, String> generatedResourcesMap = session.close();
+
+        // clear the hacky properties if set
+        if (kubernetesGroup.isPresent()) {
+            System.clearProperty(APP_GROUP_PROPERTY);
+        }
+        if (kubernetesRegistry.isPresent()) {
+            System.clearProperty(DOCKER_REGISTRY_PROPERTY);
```

Comment:
This probably needs to be done in a `try / finally` as you always want them to be cleared.

Reply:
Not I'm following. What would the `try / finally` enclose?

Reply:
I guess because of 

```
Throws:
SecurityException - if a security manager exists and its checkPropertyAccess method doesn't allow access to the specified system property.
NullPointerException - if key is null.
IllegalArgumentException - if key is empty.
```

Reply:
I pushed a change which I assume is what you wanted.